index.js

```diff
@@ -11,8 +11,10 @@ module.exports = function (options, logger) {
 
 	delete options.failOnError;
 
-	if (options.config) {
+	if (typeof options.config === "string") {
 		options = JSON.parse(fs.readFileSync(options.config));
+	} else if (typeof options.config === "object") {
+		options = options.config;
```

Comment:
Seems a bit odd to me that this should work, but the programmatic API of stylint is somewhat confusing, and your tests shows it working, so we're probably good :smi